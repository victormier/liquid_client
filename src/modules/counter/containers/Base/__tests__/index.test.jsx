import React from 'react'
import { mount, shallow } from 'enzyme'
import sinon from 'sinon'
import { shallowToJson } from 'enzyme-to-json'
import { CounterContainer } from '../index'


describe('component', () => {
  const containerProps = {
    increment: sinon.spy(),
    decrement: sinon.spy(),
    counterValue: 10,
  }

  it('renders with all props', () => {
    const container = shallow(
      <CounterContainer {...containerProps} />
    )

    expect(shallowToJson(container)).toMatchSnapshot()
  })

  it('increments counter', () => {
    const container = mount(<CounterContainer {...containerProps} />)
    container.find('button').at(0).simulate('click')
    expect(containerProps.increment.calledOnce).toEqual(true)
  })

  it('decrements counter', () => {
    const container = mount(<CounterContainer {...containerProps} />)
    container.find('button').at(1).simulate('click')
    expect(containerProps.decrement.calledOnce).toEqual(true)
  })
})
