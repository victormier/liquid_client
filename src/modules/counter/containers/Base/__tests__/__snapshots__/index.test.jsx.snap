// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component renders with all props 1`] = `
<Grid
  bsClass="container"
  componentClass="div"
  fluid={false}
>
  <PageHeader
    bsClass="page-header"
  >
    Redux Counter
  </PageHeader>
  <p>
    Lorem ipsum dolor sit amet, an eius aeque est, vel vidisse fuisset rationibus an. Eu nam ullum consul. Te putant scripta albucius pri, vim oblique scriptorem in. Eos te mazim aliquip, ea erat utinam oblique cum, has simul scripta tractatos et.
  </p>
  <p>
    Current counter value is: 
    <strong>
      10
    </strong>
  </p>
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="default"
    disabled={false}
    onClick={[Function]}
  >
    Increment
  </Button>
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="default"
    disabled={false}
    onClick={[Function]}
  >
    Decrement
  </Button>
</Grid>
`;
